### ISSUE TITLE
Query execution is very slow due to DataReader disposing taking too long

### ISSUE THREAD

**Issue by vladislav-naydenov** [dotnet/efcore#23900]
I have a simple query which runs very slow:

```c#
var photoSessions = await this.AllAsNoTracking()
                .Select(p => new PhotoSessionResponseModel
                {
                    Id = p.Id,
                    Price = p.Price,
                    Frames = p.Frames,
                    BackgroundImageUri = p.BackgroundImageUri,
                    Details = p.PhotoSessionDetails.Select(pd => new PhotoSessionDetailsResponseModel
                    {
                        Id = pd.Id,
                        Type = pd.Type,
                        LanguageCode = pd.LanguageCode,
                        Description = HtmlUtils.HtmlDecode(pd.Description)
                    }).ToList()
                })
                .ToListAsync();
```

I have enabled logging and found out that the db query gets executed very fast (10ms) but the slowness comes from disposing the DataReader which takes more than 10 seconds sometimes.

![2021-01-15](https://user-images.githubusercontent.com/3259558/104768073-bb4b5a80-5775-11eb-8e7a-1b10586a9135.png)

**Comment by smitpatel** [comment]
10ms is time to execute DbCommand and generate DbDataReader. Data reader gets disposed only after all results have been read, which can take time based on how much data is in result set. Is this happening on empty table?

**Comment by vladislav-naydenov** [comment]
It's happening on a table with 1 record.

**Comment by smitpatel** [comment]
Can you share a runnable repro code (with data) which demonstrate what you are seeing?

**Comment by vladislav-naydenov** [comment]
Hi @smitpatel,

I've created a project for reproducing the issue. This is the repo: https://github.com/vladislav-naydenov/BlazorEfCoreRepro. The file https://github.com/vladislav-naydenov/BlazorEfCoreRepro/blob/master/Server/Properties/DBPopulate.sql contains the queries for populating the db with the same data I'm using.

**Comment by ajcvickers** [comment]
@vladislav-naydenov Looks like you are running into this: https://github.com/dotnet/SqlClient/issues/593 due to loading a very long (> 1MB) string. Changing from `ToListAsync` to `ToList` drastically improves performance on my machine.

/cc @roji

**Comment by roji** [comment]
@ajcvickers yep, looks like another case of this.

**Comment by roji** [comment]
Duplicate of dotnet/SqlClient#593

**Comment by vladislav-naydenov** [comment]
@ajcvickers, @roji 

Thanks for the clarification. Will switch to using ToList until the issue is resolved
